src/api/cli/create_rawtx.go

```diff
@@ -430,36 +424,36 @@ func getKeys(wlt *wallet.Wallet, outs []UnspentOut) ([]cipher.SecKey, error) {
 	return keys, nil
 }
 
-func getSufficientUnspents(unspents []UnspentOut, coins uint64) ([]UnspentOut, error) {
-	var totalCoins uint64
-	var outs []UnspentOut
+func getSufficientUnspents(unspents *webrpc.OutputsResult, coins uint64) ([]UnspentOut, error) {
+	spendableOuts := unspents.Outputs.SpendableOutputs()
+	var spendCoins uint64
+	var spendOuts []UnspentOut
+	for i, out := range spendableOuts {
+		c, err := droplet.FromString(out.Coins)
+		if err != nil {
+			return nil, err
+		}
 
-	addrOuts := make(map[string][]UnspentOut)
-	for _, u := range unspents {
-		addrOuts[u.Address] = append(addrOuts[u.Address], u)
-	}
+		spendCoins += c
+		spendOuts = append(spendOuts, UnspentOut{spendableOuts[i]})
 
-	for _, us := range addrOuts {
-		for i, u := range us {
-			c, err := droplet.FromString(u.Coins)
-			if err != nil {
-				return nil, err
-			}
-
-			if c == 0 {
-				continue
-			}
+		if spendCoins >= coins {
+			return spendOuts, nil
+		}
+	}
 
-			totalCoins += c
-			outs = append(outs, us[i])
+	unconfirmedOuts := unspents.Outputs.IncomingOutputs
+	unconfirmedBalance, err := unconfirmedOuts.Balance()
+	if err != nil {
+		return nil, fmt.Errorf("get unconfirmed balance failed: %v", err)
+	}
 
-			if totalCoins >= coins {
-				return outs, nil
-			}
-		}
+	if unconfirmedBalance.Coins+spendCoins < coins {
```

Comment:
`if unconfirmedBalance.Coins+spendCoins > coins` ?

Reply:
oh, should rename the unconfirmedBalance to unconfirmedIncoming, and spendCoins to spendableCoins

Reply:
Can you fix it, the code seems wrong